### PR TITLE
NxCombobox Filter Variations - RSC-1244

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-1244

### CODE REVIEW

File: gallery/src/components/NxCombobox/NxComboboxPage.tsx [hunk 1]
```diff
@@ -176,6 +180,30 @@ const NxComboboxPage = () =>
...
+              <NxTable.Cell>false</NxTable.Cell>
+              <NxTable.Cell>
+                <NxP>
+                  Normally, <NxCode>NxCombobox</NxCode> uses a plain <NxCode>NxTextInput</NxCode> for its text input.
+                  However, <NxCode>NxCombobox</NxCode>'s functionality can also be useful in places where the styling of
+                  a filter input is desired – cases where <NxCode>NxFilterInput</NxCode>'s{' '}
+                  <NxCode>datalist</NxCode> prop would have formerly been used. To enable use
+                  of <NxCode>NxCombobox</NxCode> in such cases, it has a <NxCode>filterInput</NxCode> prop which causes
+                  it to use an <NxCode>NxFilterInput</NxCode> internally rather than
+                  in <NxCode>NxTextInput</NxCode>. When this prop is true, an <NxCode>NxFilterInput</NxCode> with the
```
Comment (EmilyBermudez88) [inline gallery/src/components/NxCombobox/NxComboboxPage.tsx:196]:
```suggestion
                  an <NxCode>NxTextInput</NxCode>. When this prop is true, an <NxCode>NxFilterInput</NxCode> with the
```

File: gallery/src/components/NxCombobox/NxComboboxPage.tsx [hunk 2]
```diff
@@ -176,6 +180,30 @@ const NxComboboxPage = () =>
...
+                  default "filter" icon is rendered. When this prop is instead set to "search",{' '}
+                  an <NxCode>NxFilterInput</NxCode> with a "search" icon is rendered instead.
+                </NxP>
+                <NxWarningAlert>
+                  <NxCode>NxFilterInput</NxCode> does not support validation. Therefore,
```
Comment (EmilyBermudez88) [inline gallery/src/components/NxCombobox/NxComboboxPage.tsx:201]: Out of curiosity, will it support validation in the future?

File: gallery/visualtests/nxCombobox.js [hunk 3]
```diff
@@ -162,6 +164,38 @@ describe('NxCombobox', function() {
...
+      await input.focus();
+      await getPage().keyboard.type('a');
+      await getPage().waitForSelector(dropdownButtonSelector);
+
+      await checkScreenshot(component, 300, 184);
+    });
+  });
+
+  describe('when using a search input', function() {
+    it('has a filter icon', simpleTest(searchExampleSelector));
```
Comment (EmilyBermudez88) [inline gallery/visualtests/nxCombobox.js:184]:
```suggestion
    it('has a search icon', simpleTest(searchExampleSelector));
```

File: gallery/src/components/NxCombobox/NxComboboxPage.tsx (hunk 2 above)
Comment (rpokorny) [reply gallery/src/components/NxCombobox/NxComboboxPage.tsx:201]: > Out of curiosity, will it support validation in the future?

There aren't plans for it to.  The places where a filter input would be used and the places where validation would be used would seem to be completely separate.

**Review by EmilyBermudez88 — APPROVED** [review]

**Pull request merged**